sdks/java/core/src/test/java/org/apache/beam/sdk/options/ValueProviderTest.java

```diff
@@ -243,4 +244,22 @@ public String apply(String from) {
     expectedException.expectMessage("Not called from a runtime context");
     nvp.get();
   }
+
+  private static class NonSerializable {}
+
+  private static class NonSerializableTranslator
+      implements SerializableFunction<String, NonSerializable> {
+    @Override
+    public NonSerializable apply(String from) {
+      return new NonSerializable();
+    }
+  }
+
+  @Test
+  public void testNestedValueProviderSerialize() throws Exception {
+    ValueProvider<String> rvp = StaticValueProvider.of("foo");
+    ValueProvider<NonSerializable> nvp = NestedValueProvider.of(
+        rvp, new NonSerializableTranslator());
+    ValueProvider<NonSerializable> nvpCopy = SerializableUtils.ensureSerializable(nvp);
```

Comment:
drop the LHS -- the RHS is what will throw if this is broken, and `nvpCopy` is unused.

Otherwise, LGTM